File: grida-apis/apis-core/src/main/java/org/grida/config/LogFilterRegistrar.java
package org.grida.config;

import org.grida.filter.LogFilter;
import org.springframework.boot.web.servlet.FilterRegistrationBean;
import org.springframework.context.annotation.Bean;
import org.springframework.core.Ordered;

public class LogFilterRegistrar {

    @Bean
    public FilterRegistrationBean<LogFilter> logFilter() {
        FilterRegistrationBean<LogFilter> registrationBean = new FilterRegistrationBean<>();
        registrationBean.setFilter(new LogFilter());
        registrationBean.setOrder(Ordered.HIGHEST_PRECEDENCE);

        return registrationBean;
    }
}

File: grida-core/core-api/src/main/kotlin/org/grida/support/requestlogger/LogFilterRegistrar.kt
package org.grida.support.requestlogger

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.context.annotation.Bean

class LogFilterRegistrar {

    @Bean
    fun logFilter(
        objectMapper: ObjectMapper
    ): FilterRegistrationBean<LogFilter> {
        val filterRegistration = FilterRegistrationBean<LogFilter>()
        filterRegistration.filter = LogFilter(objectMapper)
        filterRegistration.order = 0
        return filterRegistration
    }
}
